app/controllers/claims_controller.rb

```diff
@@ -2,14 +2,17 @@ class ClaimsController < BasePublicController
   include PartOfClaimJourney
   include AddressDetails
 
-  skip_before_action :send_unstarted_claimants_to_the_start, only: [:new, :create, :timeout]
+  skip_before_action :send_unstarted_claimants_to_the_start, only: [:new, :create, :timeout, :reset_claim]
   before_action :initialize_session_slug_history
   before_action :check_page_is_in_sequence, only: [:show, :update]
   before_action :update_session_with_current_slug, only: [:update]
   before_action :set_backlink_path, only: [:show]
   before_action :check_claim_not_in_progress, only: [:new]
   before_action :clear_claim_session, only: [:new]
   before_action :prepend_view_path_for_policy
+  before_action :clear_teacher_detail, only: [:reset_claim]
+  before_action :save_details_check, only: [:update]
```

Comment:
I wouldn't have `:save_details_check` in a before_action rather put it in `def update` and follow the convention there, there is that big (horrible) `if` block to also check the current `params[:slug]` which would also scope the change a little bit more.